Add icp0.io /  icp1.io / caffeine.site / caffeine.website for Internet Computer domains

sbabicz opened the pull request:
# Public Suffix List (PSL) Submission

<!--
Each PSL PR needs to have a description, rationale, indication of DNS validation and syntax checking, as well as a number of acknowledgements from the submitter. This template must be included with each PR, and the submitting party MUST provide responses to all of the elements in order to be considered.
-->

<!-- #### READ THIS FIRST ####

If you haven't yet, please read our guidelines:
https://github.com/publicsuffix/list/wiki/Guidelines#submit-the-change

Also, read them again, as many skip that part and 
get confused about why their PR is delayed or does
not get accepted when their submission didn't follow them.

A recent PR using the current template is 
https://github.com/publicsuffix/list/pull/1591, although 
the organization and description were not as substantial 
as desired, which required maintainers time to visit the 
requestor's website to further research.
Having more robust org/desc improves the PR processing 
pace due to the extra cycles not being lost to research.
For an example of what an excellent description in a PR looks like
see https://github.com/publicsuffix/list/pull/615, 
although that example uses an earlier template.
-->

### Checklist of required steps

* [x] Description of Organization
* [x] Robust Reason for PSL Inclusion
* [x] DNS verification via dig

* [x] Each domain listed in the PRIVATE section has and shall maintain at least two years remaining on registration, and we shall keep the `_psl` TXT record in place in the respective zone(s).

__Submitter affirms the following:__ 
<!--
Third-party Limits are used elsewhere, such as at Cloudflare, Let's 
Encrypt, Apple, GitLab or others, and having an entry in the PSL alters 
the manner in which those third-party systems or products treat 
a given domain name or sub-domains within it.

To be clear, it is appropriate to address how those limits impact 
your domain(s) directly with that third-party, and it is inappropriate 
to submit entries to the PSL as a means to work around those limits or 
restrictions.
-->
<!--
The purpose of the question above is to expose limit workarounds.
If there are third party limits that the PR seeks to overcome, those
must be listed within the rationale section of this request, and 
provide a good level of detail the effort that was made to work directly 
with the third part(y|ies) in attempting to address this within their 
rationale response below.
In all cases, software and services should be discouraged from use of
the PSL as a rate-limiting tool, and provide clear instructions to their
own clients, partners and users on the manner in which they can directly
request rate limit increases.
We treat the following as an attestation in the public record of the 
requesting party that they are not attempting to bypass rate limits through
the PR.
-->

 * [x] This request was _not_ submitted with the objective of working around other third-party limits.

<!--
Submitter will maintain domains in good standing or may lose section.

The ongoing trust of the PSL requires it to be free of outdated or problematic entries. In making this pull request, there is a commitment by the submitter that they are going to review and maintain their relevant section. By submitting an entry, the requestor acknowledges that their entry and section may be removed if the domain does not maintain the respective _psl entries in DNS, any domain(s) within their section fail to resolve in DNS, the domain does not get renewed, expires or is otherwise unreachable. The submitter further identifies that it is their responsibility to review their submitted section within the PSL, submitting updates or removals as their domain(s) may change over time. It is also the responsibility of the submitter to provide (and keep up to date) a reachable email address within the section, and to maintain that address as it may change over time, so that they receive notices.
-->

 * [x] The submitter acknowledges that it is their responsibility to maintain the domains within their section. This includes removing names which are no longer used, retaining the _psl DNS entry, and responding to e-mails to the supplied address. Failure to maintain entries may result in removal of individual entries or the entire section.

<!--
The guidelines describe which section to place the entry, what the 
order of commented org placement, order of sorting of entries. 
(hint: TLD then SLD, ascending sorting) Although it seems pedantic, 
the sorting and formatting rules help ensure all of the automation 
that uses the PSL operates correctly. Typically both are solved or
neither.
-->

 * [x] The [Guidelines](https://github.com/publicsuffix/list/wiki/Guidelines) were carefully _read_ and _understood_, and this request conforms to them.
 * [x] The submission follows the [guidelines](https://github.com/publicsuffix/list/wiki/Format) on formatting and sorting.

<!-- 
Sorting and formatting of the entries is outlined in the guidelines 
and non-conforming requests are one of the largest sources of delay,
so getting this right initially will aid successfully having it 
proceed. Mislocated entries and trailing spaces should be avoided.
-->

<!--
In your submission, please include a role-based email address (e.g. security@example.com) rather than a personal email address (e.g. jane.doe@example.com) under your organization name, so that we can maintain contact with your organization, independent of personnel changes.
This email address will be used for any required verification or inquiries regarding your PSL listing. This inbox must be actively maintained and monitored for future communications from the PSL project for as long as the domain remains in the PSL. Any PSL inquiries sent to this address must receive a response within 30 days, as maintaining timely communication is required for continued inclusion in the PSL.
-->

 * [x] A role-based email address has been used and this inbox is actively monitored with a response time of no more than 30 days.
Contact email: `domains@dfinity.org`
**Abuse Contact:**

<!--
Please confirm that you have accessible abuse contact information on your website. 

At a minimum, you must provide an abuse contact either in the form of an email address or a web form that can be used to report abuse. This contact should be easily accessible to allow concerned parties to notify the registry or subdomain operator directly when malicious activities such as phishing, malware, or abuse are detected. For example, if you provide subdomains at example.com, where users may register subdomains such as clientname.example.com, then in case of abuse, reporters should be able to visit example.com and easily find the relevant abuse contact information.
-->

* [x] Abuse contact information (email or web form) is available and easily accessible.

  URL where abuse contact or abuse reporting form can be found: https://dfinity.org/bug-bounty
  <!-- Provide the URL where an Internet user can access the abuse contact information -->

---

For PRIVATE section requests that are submitting entries for domains that match their organization website's primary domain, please understand that this can have impacts that may not match the desired outcome and take a long time to rollback, if at all.

To ensure that requested changes are entirely intentional, make sure that you read the affectation and propagation expectations, that you understand them, and confirm this understanding. 

PR Rollbacks have lower priority, and the volunteers are unable to control when or if browsers or other parties using the PSL will refresh or update.

<!-- 
Seriously, carefully read the downline flow of the PSL and the 
guidelines. Your request could very likely alter the cookie and 
certificate (as well as other) behaviours on your core domain name in 
ways that could be problematic for your business.

Rollbacks are really not predictable, as those who use or incorporate 
the PSL do what they do, and when. It is not within the PSL volunteers' 
control to do anything about that. 

The volunteers are busy with new requests, and rollbacks are lowest 
priority, so if something gets broken by your PR, it will potentially 
stay that way for an indefinite period of time (typically long).
-->

(Link: [about propagation/expectations](https://github.com/publicsuffix/list/wiki/Guidelines#appropriate-expectations-on-derivative-propagation-use-or-inclusion))

 * [x] *Yes, I understand*. I could break my organization's website cookies and cause other issues, and the rollback timing is acceptable. *Proceed anyways*.
---

<!--
As you complete each item in the checklist please mark it with an X.

For example:
* [x] Description of Organization
-->

## Description of Organization
Request submitted by Slawomir Babicz, Infrastructure Security Engineer at Dfinity Foundation. 

Dfinity is a not-for-profit organization, a major contributor to the Internet Computer blockchain. The Internet Computer is a general-purpose blockchain that hosts canister smart contracts. It is designed to provide a World Computer that can replace traditional IT and host a new generation of Web3 services and applications that run solely from the blockchain, without the need for traditional IT. It can also play the role of Web3 orchestrator, by interacting with traditional blockchains.

**Organization Website:**
https://dfinity.org/ , https://internetcomputer.org/

## Reason for PSL Inclusion

- Content deployed on `icp0.io`, and `icp1.io` consists of unique subdomains represented as 28-byte hashes. Each subdomain corresponds to a separate website with distinct owners. Due to exposing ICP to the world, some individuals may create inappropriate content, but the community and the DFINITY Foundation actively monitor and filter such content.

- Inclusion in the PSL will help treat subdomains as separate entities, effectively isolating cookies and sensitive information between different websites.

- Inclusion in the PSL enhances the reputation and credibility of `icp0.io`, and `icp1.io` domains. It demonstrates their commitment to industry standards and best practices, promoting trust among users, businesses, and organizations interacting with these domains.

- Including `icp0.io`, `icp-api.io`, `id.ai` and `icp1.io` in the PSL will enhance trust and bolster security measures.

- Domains `caffeine.site`, `caffeine.ai` will be utilized specifically to showcase decentralized applications (dAPPs) built on the ICP blockchain, highlighting innovation and development within the ICP ecosystem.

**Number of users this request is being made to serve:**
1000000

## DNS Verification
```
❯ dig TXT +short _psl.icp0.io @1.1.1.1
"https://github.com/publicsuffix/list/pull/2424"
❯ dig TXT +short _psl.raw.icp0.io @1.1.1.1
"https://github.com/publicsuffix/list/pull/2424"
❯ dig TXT +short _psl.raw.ic0.app @1.1.1.1
"https://github.com/publicsuffix/list/pull/2424"
❯ dig TXT +short _psl.icp1.io @1.1.1.1
"https://github.com/publicsuffix/list/pull/2424"
❯ dig TXT +short _psl.caffeine.site @1.1.1.1
"https://github.com/publicsuffix/list/pull/2424"
❯ dig TXT +short _psl.raw.icp1.io @1.1.1.1
"https://github.com/publicsuffix/list/pull/2424"
❯ dig TXT +short _psl.icp-api.io @1.1.1.1
"https://github.com/publicsuffix/list/pull/2424"
❯ dig _psl.caffeine.ai TXT +short
"https://github.com/publicsuffix/list/pull/2424"
❯ dig _psl.id.ai TXT +short
"https://github.com/publicsuffix/list/pull/2424"
```

hiifeng commented:
Is this a draft? But it is open. After testing, the domains are sorted correctly. But no TXT verification record has been added. Please refer to the following document to add TXT verification record for the domain as soon as possible.

https://github.com/publicsuffix/list/wiki/Guidelines#rfc8553-dns-authentication

groundcat commented:
Yes, you would need to add a TXT record for `_psl.raw.icp0.io` specifically, even if you've already demonstrated ownership of `icp0.io`. Each domain entry in the PSL needs its own verification record. You should also keep these DNS records in place for as long as you want your entries to remain in the PSL. The project uses these records to verify ownership, and future automation might remove entries where the verification record is no longer present.

hiifeng commented:
> Thanks, I have added all required DNS entries.

I have verified all the relevant domains and all have correctly set up TXT verification records.
But if you can modify the PR and add the "+short" parameter to facilitate other people's viewing, it will be more perfect.

```
root@hcss-ecs-c37d:~# dig TXT +short _psl.ic0.app
"https://github.com/publicsuffix/list/pull/2424"
root@hcss-ecs-c37d:~# dig TXT +short _psl.raw.ic0.app
"https://github.com/publicsuffix/list/pull/2424"
root@hcss-ecs-c37d:~# dig TXT +short _psl.icp0.io
"https://github.com/publicsuffix/list/pull/2424"
root@hcss-ecs-c37d:~# dig TXT +short _psl.raw.icp0.io
"https://github.com/publicsuffix/list/pull/2424"
root@hcss-ecs-c37d:~# dig TXT +short _psl.icp1.io
"https://github.com/publicsuffix/list/pull/2424"
root@hcss-ecs-c37d:~# dig TXT +short _psl.raw.icp1.io
"https://github.com/publicsuffix/list/pull/2424"
root@hcss-ecs-c37d:~# dig TXT +short _psl.caffeine.site
"https://github.com/publicsuffix/list/pull/2424"
root@hcss-ecs-c37d:~# dig TXT +short _psl.caffeine.website
"https://github.com/publicsuffix/list/pull/2424"
root@hcss-ecs-c37d:~#
```

In addition, I noticed that the content of "Number of users this request is being made to serve:" is empty, please fill it in as soon as possible.

sbabicz commented:
Added DNS records with +short and users estimate.

hiifeng commented:
> Added DNS records with +short and users estimate.

One last question, as shown in the picture below, please select one of them, if you are not trying to solve other third-party restrictions. Please delete the upper part.
After completion, I will help you generate an acceptance checklist.

<img width="564" alt="1743748929801" src="https://github.com/user-attachments/assets/de8cac85-12a7-4c7f-8b0e-58e53bfe1f94" />

sbabicz commented:
@hiifeng thanks! 
I have removed the section "We are listing any ... " as thats not our purpose here.

hiifeng commented:
> @hiifeng thanks! I have removed the section "We are listing any ... " as thats not our purpose here.

Sorry, I should have checked all the issues and told you everything at once. I'm a beginner and have only recently started volunteering to help review PRs, so I was a bit flustered.

Please renew ic0.app, caffeine.site, and caffeine.website for another two years to meet the acceptance requirements. Below are the expiration dates I found:

ic0.app – 2025-05-21
icp0.io – 2030-09-06
icp1.io – 2030-09-06
caffeine.site – 2025-09-20
caffeine.website – 2026-03-05

sbabicz commented:
Hey, appreciate your help!
I am interacting in the background with different teams making that adjustments, so its also taking time. 

I will be planning to add one more additional domain, and will request extending the expiration date (currently set to auto-renew).

sbabicz commented:
```
❯ whois ic0.app | grep "Registry Expiry";
Registry Expiry Date: 2025-05-21T18:50:35Z  <-- awaiting domain expiration extension

❯ whois icp0.io | grep "Registry Expiry";
Registry Expiry Date: 2030-09-06T18:17:03Z

❯ whois icp1.io | grep "Registry Expiry";
Registry Expiry Date: 2030-09-06T18:17:03Z

❯ whois icp-api.io | grep "Registry Expiry";
Registry Expiry Date: 2028-02-15T20:44:12Z

❯ whois caffeine.site | grep "Registry Expiry";
Registry Expiry Date: 2025-09-20T23:59:59.0Z <-- awaiting domain expiration extension

❯ whois caffeine.website | grep "Registry Expiry";
Registry Expiry Date: 2026-03-05T23:59:59.0Z <-- awaiting domain expiration extension
```

hiifeng commented:
> ```
> ❯ whois ic0.app | grep "Registry Expiry";
> Registry Expiry Date: 2025-05-21T18:50:35Z  <-- awaiting domain expiration extension
> 
> ❯ whois icp0.io | grep "Registry Expiry";
> Registry Expiry Date: 2030-09-06T18:17:03Z
> 
> ❯ whois icp1.io | grep "Registry Expiry";
> Registry Expiry Date: 2030-09-06T18:17:03Z
> 
> ❯ whois icp-api.io | grep "Registry Expiry";
> Registry Expiry Date: 2028-02-15T20:44:12Z
> 
> ❯ whois caffeine.site | grep "Registry Expiry";
> Registry Expiry Date: 2025-09-20T23:59:59.0Z <-- awaiting domain expiration extension
> 
> ❯ whois caffeine.website | grep "Registry Expiry";
> Registry Expiry Date: 2026-03-05T23:59:59.0Z <-- awaiting domain expiration extension
> ```

As shown in your query results, the domains ic0.app, caffeine.site, and caffeine.website currently do not meet the requirements.
Could you please resolve the domain renewal issue as soon as possible?

Item 7 in Validation and Non-Acceptance Factors states:

> We now require that domains submitted as private section entries have expiration dates more than 2 years beyond the submitting date of a PR. Please include a statement within the rationale that a] the domain name(s) submitted have at least 2 years of registration period left in them at the time of submitting the name, and b] include a commitment that you will maintain their registration in good standing with more than a year left in their term.

https://github.com/publicsuffix/list/wiki/Guidelines#validation-and-non-acceptance-factors

sbabicz commented:
Yes, this has already been escalated to the team responsible for domain management. Apologies for the current state of the affected domains — we’re aware that `ic0.app`, `caffeine.site`, and `caffeine.website` do not yet meet the renewal requirements. Renewal requests have been submitted to extend their expiration by an additional two years to ensure compliance. We'll update the PR once the extensions are confirmed. Thanks for your patience!

sbabicz commented:
```
❯ whois caffeine.site | grep "Registry Expiry"
Registry Expiry Date: 2027-09-20T23:59:59.0Z
```

```
❯ whois caffeine.website | grep "Registry Expiry";
Registry Expiry Date: 2027-03-05T23:59:59.0Z
```

simon-friedberger commented:
caffeine.website expires 2027-03-05

Reminder: The expiration must at all times remain above two years. Not following the rules can lead to removal from the list at any time.

sbabicz commented:
Removed two domains: icp0.io and caffeine.website  from the list as extending the domain could not be done on time.

hiifeng commented:
- [x]  Expiration (Note: Must STAY >2y at all times)
   - [x] ic0.app `2025-05-21`
   - [x] icp0.io `Registry Expiry Date: 2030-09-06T18:17:03Z`
   - [x] icp1.io `Registry Expiry Date: 2030-09-06T18:17:03Z`
   - [x] icp-api.io `Registry Expiry Date: 2028-02-15T20:44:12Z`
   - [x] caffeine.site `Registry Expiry Date: 2027-09-20T23:59:59.0Z` 
   - [x] caffeine.ai `Registry Expiry Date: 2034-07-20T21:17:31Z`
   - [x] id.ai `Registry Expiry Date: 2030-02-02T01:25:28Z`
- [x]  DNS _psl entries (Note: Must STAY in place)
- [x]  Sorting
- [x]  Reasoning/Organization description
  - The reasons for inclusion in the report and the number of service users appear to meet the requirements.
- [x]  Non-personal email address
- [ ]  Abuse contact
  - I noticed that ic0.app, icp0.io, icp1.io, and icp-api.io are all redirected to https://dashboard.internetcomputer.org, but there is no abuse contact information or abuse report form URL available on that page. caffeine.site and id.ai do not have any web services.
  - I suggest redirecting all domains to https://dfinity.org, where the abuse contact information is clearly visible.

sbabicz commented:
caffeine.site and id.ai are domains to be deployed, and there will be information where to publish abuse report. 
Abuse can be sent to the dns registrar as there are not yet associated web services yet.

hiifeng commented:
> caffeine.site and id.ai are domains to be deployed, and there will be information where to publish abuse report. Abuse can be sent to the dns registrar as there are not yet associated web services yet.

Please pay attention to the validity period of ic0.app. Everything else looks good to me.

hiifeng commented:
> I have fixed formatting, all domains in the PR meet validity time

Has `ic0.app` been renewed? The expiration date still shows as `2025-05-21T18:50:35.406Z` when I check.

sbabicz commented:
> > I have fixed formatting, all domains in the PR meet validity time
> 
> Has `ic0.app` been renewed? The expiration date still shows as `2025-05-21T18:50:35.406Z` when I check.

I have removed the ic0.app from the file as we cannot get the extension yet. Will be added in another PR once extended.

simon-friedberger commented:
Can you provide further information on where this user estimate comes from?

simon-friedberger commented:
Why would the number of users (those are people) match the number of smart contracts?

sbabicz commented:
Thank you for the question — you're right that the number of smart contracts doesn't directly equal the number of users.

The 1,000,000 user estimate is a conservative approximation based on the cumulative reach of applications (dApps) deployed as canisters on the Internet Computer. While each canister may serve a different purpose, many public-facing dApps (such as OpenChat) can reach tens or hundreds of thousands of users individually.

Importantly, DFINITY itself does not track user counts, as it provides the underlying Internet Computer protocol, not the applications. The platform is decentralized by design — developers deploy their own applications and manage their own user bases independently.

As such, exact user numbers are difficult to determine centrally.

This is why Public Suffix List inclusion is important: it ensures proper security and isolation between these independently operated subdomains.

slawomirbabicz commented:
Is there anything else I should address? From my perspective the PR is ready to be merged.

simon-friedberger commented:
You have to fix the failing tests, please.

sbabicz commented:
Fixed sorting by running `go run ./psltool fmt ../public_suffix_list.dat`
